torch_xla/csrc/ops/cumprod.cpp

```diff
@@ -0,0 +1,50 @@
+#include "torch_xla/csrc/ops/cumprod.h"
+
+#include "torch_xla/csrc/convert_ops.h"
+#include "torch_xla/csrc/helpers.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+#include "torch_xla/csrc/reduction.h"
+#include "torch_xla/csrc/torch_util.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::XlaOp LowerCumProd(const xla::XlaOp& input, xla::int64 dim,
+                        c10::optional<at::ScalarType> dtype) {
+  xla::XlaOp casted_input = CastToScalarType(input, dtype);
+  xla::Shape input_shape = XlaHelpers::ShapeOfXlaOp(casted_input);
+  xla::XlaOp init = XlaHelpers::ScalarValue<float>(
+      1, input_shape.element_type(), casted_input.builder());
+  xla::XlaComputation reducer =
+      XlaHelpers::CreateMulComputation(input_shape.element_type());
+  return BuildCumulativeComputation(casted_input, dim, reducer, init);
+}
+
+}  // namespace
+
+CumProd::CumProd(const Value& input, xla::int64 dim,
+                 c10::optional<at::ScalarType> dtype)
+    : Node(ir::OpKind(at::aten::cumprod), {input}, input.shape(),
```

Comment:
Hmm, this is prolly not correct. We are reducing, right?
So maybe the reduce dimensions gets squashed?

Reply:
Nope, nothing gets squashed since this is cumulative. Shape remains the same, with the exception of the element type -- this is the part which is actually incorrect.

Reply:
Done.